src/mobile-ui-react/MobileUi.tsx

```diff
@@ -5,13 +5,21 @@
 import * as React from "react";
 import { BackendError, Localization } from "@itwin/core-common";
 import { getCssVariable, getCssVariableAsNumber } from "@itwin/core-react";
-import { ColorTheme, SessionStateActionId, SyncUiEventDispatcher, SyncUiEventId, SYSTEM_PREFERRED_COLOR_THEME, UiFramework, UiSyncEventArgs } from "@itwin/appui-react";
+import { ColorTheme, SessionStateActionId, SyncUiEventDispatcher, SyncUiEventId, SYSTEM_PREFERRED_COLOR_THEME, UiFramework } from "@itwin/appui-react";
 import { EmphasizeElements, IModelApp, IModelConnection, ScreenViewport, SelectionSet, Tool, Viewport } from "@itwin/core-frontend";
 import { BeEvent, BeUiEvent, BriefcaseStatus, Id64Set, Listener } from "@itwin/core-bentley";
 import { getAllViewports, getEmphasizeElements, Messenger, MobileCore, UIError } from "@itwin/mobile-sdk-core";
 
 import "./MobileUi.scss";
 
+// NOTE: All of the below is what AppUI says we need to do to avoid using the deprecated
+// UiSyncEventArgs type from AppUI. We can't do this in one file and reference from others,
+// because that would require US to export UiSyncEventArgs, and we definitely don't want to do that.
+// So every file where we need to use this type, we must include all three of the below lines.
+type ListenerType<TEvent extends BeEvent<Listener>> = TEvent extends BeEvent<infer TListener> ? TListener : never;
+type UiSyncEventListener = ListenerType<typeof SyncUiEventDispatcher.onSyncUiEvent>;
+type UiSyncEventArgs = Parameters<UiSyncEventListener>[0];
```

Comment:
Rather than the 3 lines, copied everywhere, I suggest a single new reusable `ListenerArgs` type:

```suggestion
/** Helper type for SyncUi events */
export type ListenerArgs<TEvent extends BeEvent<Listener>> = TEvent extends BeEvent<infer TListener> ? Parameters<TListener>[0] : never;
```